examples/AdvancedOperations/AddAdCustomizer.php

```diff
@@ -0,0 +1,529 @@
+            'string_value' => new StringValue(['value' => $price])
+        ]);
+                
+        $dateAttributeValue = new FeedItemAttributeValue([
+            'feed_attribute_id' => $adCustomizerFeedAttributes['Date'],
+            'string_value' => new StringValue(['value' => $date])
+        ]);
+
+        $feedItem = new FeedItem([
+            'feed' => $adCustomizerFeedResourceName,
+            'attribute_values' => [
+                $nameAttributeValue,
+                $priceAttributeValue,
+                $dateAttributeValue
+            ]
+        ]);
+
+        $feedItemOperation = new FeedItemOperation();
+        $feedItemOperation->setCreate($feedItem);
+
+        return $feedItemOperation;
+    }
+
+  /**
+   * Restricts the feed items to work only with a specific ad group; this prevents the feed items
+   * from being used elsewhere and makes sure they are used only for customizing a specific ad
+   * group.
+   *
+   * @param GoogleAdsClient googleAdsClient the Google Ads API client
+   * @param int $customerId the customer ID
+   * @param array adGroupIds the ad group IDs to bind the feed items to.
+   * @param array feedItemResourceNames the resource names of the feed items.
+   */
+    private function createFeedItemTargets(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        array $adGroupIds,
+        array $feedItemResourceNames
+    ) {
+        // Bind each feed item to a specific ad group to make sure it will only be used to customize
+        // ads inside that ad group; using the feed item elsewhere will result in an error.
+        for ($i = 0; $i < count($feedItemResourceNames); $i++) {
+            $feedItemResourceName = $feedItemResourceNames[$i];
+            $adGroupId = $adGroupIds[$i];
+
+            $feedItemTarget = new FeedItemTarget([
+                "feed_item" => new StringValue(["value" => $feedItemResourceName]),
+                "ad_group" => ResourceNames::forAdGroup($customerId, $adGroupId)
+            ]);
+
+            // Creates the operation.
+            $feedItemTargetOperation = new FeedItemTargetOperation();
+            $feedItemTargetOperation->setCreate($feedItemTarget);
+
+            // Issues a mutate request to add the feed item target.
+            $feedItemTargetServiceClient = $googleAdsClient->getFeedItemTargetServiceClient();
+            $feedItemTargetResponse = $feedItemTargetServiceClient->mutateFeedItemTargets(
+                $customerId,
+                [$feedItemTargetOperation]
+            );
+
+            $feedItemTargetResourceName =
+                $feedItemTargetResponse->getResults()[0]->getResourceName();
+            printf("Added feed item target named '%s'.%s", $feedItemTargetResourceName, PHP_EOL);
+        }
+    }
+
+    /**
+     * Creates expanded text ads that use the ad customizer feed to populate the placeholders.
+     *
+     * @param GoogleAdsClient googleAdsClient the Google Ads API client.
```

Comment:
No dot at the end please.
Please check this throughout your file.

Reply:
Done.